### PR TITLE
protocol: add initial block hash field to the Chain object

### PR DESCRIPTION
**Pull request by bobg**
Resurrects some code from https://github.com/chain/chain/pull/26. Related to https://github.com/chain/chainprv/issues/2404.

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

**Comment by jbowens** [comment]
lgtm

File: protocol/block_test.go [hunk 1]
```diff
@@ -221,16 +221,19 @@ func TestValidateBlockForSig(t *testing.T) {
 // It commits b1 before returning.
 func newTestChain(tb testing.TB, ts time.Time) (c *Chain, b1 *bc.Block) {
 	ctx := context.Background()
-	c, err := NewChain(ctx, memstore.New(), mempool.New(), nil)
+
+	var err error
+
+	b1, err = NewInitialBlock(nil, 0, ts)
```
Comment (kr) [inline protocol/block_test.go]: nit: we could use `:=` here

File: protocol/protocol.go [hunk 2]
```diff
@@ -103,16 +103,18 @@ type Pool interface {
 // validation logic from package validation to decide what
 // objects can be safely stored.
 type Chain struct {
+	InitialBlockHash  bc.Hash
```
Comment (kr) [inline protocol/protocol.go]: We might consider making this a method to enforce that it's readonly.

**Comment by kr** [comment]
(I was hitting the approve button anyway, so I my comments above are not a big deal.)

**Pull request merged**